inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp

```diff
@@ -376,6 +376,45 @@ void FrontEnd::parseLayer(const Model& model, const ie::CNNLayerPtr& layer, cons
     }
 }
 
+void FrontEnd::processTrivialCases(const Model& model) {
+    DataVector trivialInputs, trivialOutputs;
+
+    for (const auto& data : model->datas()) {
+        if (data->usage() == DataUsage::Input || data->usage() == DataUsage::Intermediate || data->usage() == DataUsage::Const) {
+            if (data->numConsumers() == 0 && data->childDataToShapeEdges().size() == 0) {
+                trivialInputs.push_back(data);
+            }
+        }
+        if (data->usage() == DataUsage::Output || data->usage() == DataUsage::Intermediate) {
+            if (data->producer() == nullptr && data->childDataToShapeEdges().size() == 0) {
+                trivialOutputs.push_back(data);
+            }
+        }
+    }
+
+    VPU_THROW_UNLESS(trivialOutputs.size() <= trivialInputs.size(),
+                     "Number of trivial outputs should not be more than number of trivial input/const data objects,"
+                     "but got {} and {} respectively", trivialOutputs.size(), trivialInputs.size());
+
+    for (size_t i = 0; i < trivialOutputs.size(); i++) {
+        const auto trivialInputIt = std::find_if(trivialInputs.cbegin(), trivialInputs.cend(), [&trivialOutputs, i](const Data& trivialInput) {
```

Comment:
Such an approach assumes there are several vpu data objects for single IE data object. Is it more reasonable to store that information in a dedicated data structure (e.g. something like a map from IE data objects to collection of vpu data objects)? I guess one of key moments is we're interested only in cases when there are more than one element in value collection

Reply:
This is the only place where such structure may be used, so I'm not sure it's reasonable for now. `std::find_if` do the job without any additional structures.

Reply:
But does it have the same efficiency?

Reply:
Your suggestion is much more efficient, done.